hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -1044,28 +1053,64 @@ int SaraNcpClient::selectSimCard(ModemState& state) {
     }
 
     if (ncpId() == PLATFORM_NCP_SARA_R410) {
-        // Set UMNOPROF = SIM_SELECT
-        auto resp = parser_.sendCommand("AT+UMNOPROF?");
-        reset = false;
-        int umnoprof = static_cast<int>(UbloxSaraUmnoprof::NONE);
-        auto r = CHECK_PARSER(resp.scanf("+UMNOPROF: %d", &umnoprof));
-        CHECK_PARSER_OK(resp.readResult());
-        if (r == 1 && static_cast<UbloxSaraUmnoprof>(umnoprof) == UbloxSaraUmnoprof::SW_DEFAULT) {
-            // Disconnect before making changes to the UMNOPROF
-            r = CHECK_PARSER(parser_.execCommand("AT+COPS=2,2"));
-            if (r == AtResponse::OK) {
+        int resetCount = 0;
+        int isTwilio = 0;
+        // Check if we are using a Twilio Super SIM based on ICCID
+        char buf[32] = {};
+        auto lenCcid = getIccidImpl(buf, sizeof(buf));
+        CellularNetworkConfig netConfig;
+        CHECK_TRUE(lenCcid > 5, SYSTEM_ERROR_AT_NOT_OK);
+        netConfig = networkConfigForIccid(buf, lenCcid);
+        if (!strcmp(netConfig.apn(), "super")) { // if Twilio Super SIM
+            isTwilio = 1;
+        }
+        do {
+            // Set UMNOPROF = SIM_SELECT
+            auto resp = parser_.sendCommand("AT+UMNOPROF?");
+            reset = false;
+            int umnoprof = static_cast<int>(UbloxSaraUmnoprof::NONE);
+            auto r = CHECK_PARSER(resp.scanf("+UMNOPROF: %d", &umnoprof));
+            CHECK_PARSER_OK(resp.readResult());
+            if (r == 1 && static_cast<UbloxSaraUmnoprof>(umnoprof) == UbloxSaraUmnoprof::SW_DEFAULT) {
+                // Disconnect before making changes to the UMNOPROF
+                auto respCfun = parser_.sendCommand(UBLOX_CFUN_TIMEOUT, "AT+CFUN?");
```

Comment:
Since we have the ability to check, and repeatedly setting CFUN=0,0 is somewhat of an unknown operation behind the scenes, I think checking first is a bit easier on the system.  And it takes very little time to do.